The new GNU linker gold requires -lcrypto to be specified explicitly.

Index: fdm-1.6/GNUmakefile
===================================================================
--- fdm-1.6.orig/GNUmakefile	2009-04-14 06:00:51.000000000 +0200
+++ fdm-1.6/GNUmakefile	2009-04-14 06:00:58.000000000 +0200
@@ -66,7 +66,7 @@
 DEFS+= $(shell getconf LFS_CFLAGS) -D_GNU_SOURCE \
        -DNO_STRLCPY -DNO_STRLCAT -DNO_STRTONUM -DNO_SETPROCTITLE \
        -DNO_QUEUE_H -DNO_TREE_H
-LIBS+= -lresolv
+LIBS+= -lresolv -lcrypto
 # Required for LLONG_MAX and friends
 CFLAGS+= -std=c99
 endif
